src/core/pull/index.ts

```diff
@@ -0,0 +1,154 @@
+/*!
+ * V4Fire Core
+ * https://github.com/V4Fire/Core
+ *
+ * Released under the MIT license
+ * https://github.com/V4Fire/Core/blob/master/LICENSE
+ */
+
+/**
+ * [[include:core/pull/README.md]]
+ * @packageDocumentation
+ */
+
+import { SyncPromise } from 'core/prelude/structures';
+
+interface SpecialSettings<T>{
+	maxSize:number;
+	onTake(value:T, pull:Pull<T>, args:any): void;
+	onFree(value:T, pull:Pull<T>, args:any): void;
+}
+
+interface ReturnType<T>{
+	free(val:T, args:any): void;
+	value:T;
+}
+
+/**
+ * Simple implementation of pull with stack
+ * @typeparam T - pull element
+ */
+export default class Pull<T> {
+
+	/**
+	 * Amount of available now objects
+	 */
+	available: number;
+
+	/**
+	 * Data structure that contain pull's object
+	 */
+	stack:T[]=[];
+
+	/**
+	 * Hook that are activated before (free) function
+	 * @param value value from free(value)
+	 * @param pull this pull
+	 * @param args params that are given in free(value,...args)
+	 */
+	onFree:(value: T, pull: Pull<T>, args: any)=> void;
+
+	/**
+	 * Hook that are activated before this.take or this.takeOrCreate
```

Comment:
После описания обязательно должен быть перевод строки, чтобы визуально не слипалось

```suggestion
	 * Hook that are activated before this.take or this.takeOrCreate
	 *
```